### PR TITLE
feat(ButtonLinkDanger): create component

### PR DESCRIPTION
**Pull request by marcoskolodny**
Issue: [Link](https://jira.tid.es/browse/WEB-1626)
Specs: [Link](https://www.figma.com/file/koROdh3HpEPG2O8jG52Emh/%F0%9F%94%B8-Buttons-Component-Specs?type=design&node-id=3448-10628&mode=design&t=ivVRqI9cucIxG8BE-0)

### CODE REVIEW

File: src/button.tsx
Comment (marcoskolodny) [inline src/button.tsx:1]: If you use `Hide whitespace` you can review changes in a more friendly way

File: src/button.tsx [hunk 1]
```diff
@@ -475,134 +477,163 @@ interface ButtonLinkToProps extends ButtonLinkCommonProps {
...
-    const [isOnPressPromiseResolving, setIsOnPressPromiseResolving] = React.useState(false);
-
-    const showSpinner = props.showSpinner || isOnPressPromiseResolving;
-    const showChevron = props.withChevron ?? (!!props.href || !!props.to);
+export const BaseButtonLink = React.forwardRef<TouchableElement, ButtonLinkProps & {type: ButtonLinkType}>(
+    ({type, ...props}, ref) => {
+        const {formStatus} = useForm();
+        const isInverse = useIsInverseVariant();
+        const {textPresets} = useTheme();
+        const {eventFormat} = useTrackingConfig();
```
Comment (marcoskolodny) [inline src/button.tsx:485]: Unrelated change: eventFormat should be imported by `useTrackingConfig()`, not from `useTheme()`

File: src/button.tsx [hunk 2]
```diff
@@ -475,10 +476,14 @@ interface ButtonLinkToProps extends ButtonLinkCommonProps {
...
-export const ButtonLink = React.forwardRef<TouchableElement, ButtonLinkProps>((props, ref) => {
+export const BaseButtonLink = React.forwardRef<
+    TouchableElement,
+    ButtonLinkProps & {type: ButtonLinkType; withChevron?: boolean}
+>(({type, ...props}, ref) => {
     const {formStatus} = useForm();
     const isInverse = useIsInverseVariant();
-    const {analytics, textPresets} = useTheme();
+    const {textPresets} = useTheme();
+    const {eventFormat} = useTrackingConfig();
```
Comment (marcoskolodny) [inline src/button.tsx:486]: Unrelated change: `eventFormat` should be imported by `useTrackingConfig()`, not from `useTheme()`

**Review by yceballost — APPROVED** [review]
@AnaMontes11 textLinkDangerInverse is not needed. Remove from specs pls!

**Comment by AnaMontes11** [comment]
> @AnaMontes11 textLinkDangerInverse is not needed. Remove from specs pls!

Specs updated.

File: src/__screenshot_tests__/__image_snapshots__/button-screenshot-test-tsx-buttons-link-button-danger-small-1-snap.png
Comment (pladaria) [inline src/__screenshot_tests__/__image_snapshots__/button-screenshot-test-tsx-buttons-link-button-danger-small-1-snap.png:1]: shouldn't this be small?

Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/button-screenshot-test-tsx-buttons-link-button-danger-small-1-snap.png:1]: actually, ButtonLink and ButtonLinkDanger don't have small version. These screenshots shouldn't exist, I've removed them

File: src/button.tsx (hunk 2 above)
Comment (atabel) [reply src/button.tsx:486]: remember to close the related jira ticket with this bug

File: src/button.tsx [hunk 3]
```diff
@@ -475,10 +476,15 @@ interface ButtonLinkToProps extends ButtonLinkCommonProps {
 
 export type ButtonLinkProps = ButtonLinkOnPressProps | ButtonLinkHrefProps | ButtonLinkToProps;
 
-export const ButtonLink = React.forwardRef<TouchableElement, ButtonLinkProps>((props, ref) => {
+export const BaseButtonLink = React.forwardRef<
```
Comment (atabel) [inline src/button.tsx:479]: this shouldn't be exported

**Review by atabel — APPROVED** [review]

File: src/button.tsx (hunk 3 above)
Comment (marcoskolodny) [reply src/button.tsx]: good catch

**Review by pladaria — APPROVED** [review]

**Review by AnaMontes11 — APPROVED** [review]
